src/commands/dev/tls/certs.rs

```diff
@@ -0,0 +1,88 @@
+use openssl::asn1::Asn1Time;
+use openssl::bn::{BigNum, MsbOption};
+use openssl::hash::MessageDigest;
+use openssl::pkey::PKey;
+use openssl::rsa::Rsa;
+use openssl::x509::extension::{BasicConstraints, KeyUsage, SubjectKeyIdentifier};
+use openssl::x509::{X509NameBuilder, X509};
+use std::fs;
+use std::path::PathBuf;
+
+use crate::settings::get_wrangler_home_dir;
+use crate::terminal::message;
+
+/// Create files for cert and private key
+fn create_output_files() -> Result<Option<(PathBuf, PathBuf)>, failure::Error> {
+    let home = get_wrangler_home_dir()?.join("config");
+    let cert = home.join("dev-cert.pem");
+    let privkey = home.join("dev-privkey.rsa");
+
+    if cert.exists() && privkey.exists() {
+        Ok(None)
+    } else {
+        fs::create_dir_all(&home)?;
+
+        message::info(format!("Generating certificate and private key for https server, if you would like to use your own you can replace `dev-cert.pem` and `dev-privkey.rsa` at {}", home.to_str().unwrap()).as_str());
+
+        Ok(Some((cert, privkey)))
+    }
+}
+
+/// Generate cert and private key for local https server
+pub fn generate_cert() -> Result<(), failure::Error> {
+    let files = create_output_files()?;
+    if files.is_none() {
+        return Ok(());
+    }
+
+    let (cert_file, priv_file) = files.unwrap();
+
+    let rsa = Rsa::generate(2048)?;
+    let privkey = PKey::from_rsa(rsa)?;
+
+    let mut x509_name = X509NameBuilder::new()?;
+    x509_name.append_entry_by_text("C", "US")?;
+    x509_name.append_entry_by_text("ST", "TX")?;
+    x509_name.append_entry_by_text("O", "Example")?;
+    x509_name.append_entry_by_text("CN", "example")?;
```

Comment:
what's happening here?

Reply:
So this and the previous comment are the [attributes](https://docs.oracle.com/cd/E24191_01/common/tutorials/authz_cert_attributes.html) for the certificate. Usually they're the name of the company/organization for the certificate authority or who created the certificate. I was just using Wrangler or example.